File: pkg/dto/project_member.go
package dto

import "github.com/KubeOperator/KubeOperator/pkg/model"

type ProjectMember struct {
	model.ProjectMember
	UserName string `json:"userName"`
}

type ProjectMemberOP struct {
	Operation string                `json:"operation" validate:"required"`
	Items     []ProjectMemberCreate `json:"items" validate:"required"`
}

type ProjectMemberCreate struct {
	ProjectName string `json:"projectName" validate:"required"`
	Role        string `json:"role" validate:"oneof=CLUSTER_MANAGER PROJECT_MANAGER"`
	Username    string `json:"username" validate:"required"`
}

type AddMemberResponse struct {
	Items []string `json:"items"`
}

File: ui/src/app/business/project/project-member/project-member.ts
import {BaseModel, BaseRequest} from '../../../shared/class/BaseModel';

export class ProjectMember extends BaseModel {
    userName: string;
    role: string;
    email: string;
}

export class ProjectMemberRequest extends BaseRequest {
    name: string;
    role: string;
    projectId: string;
}

export class ProjectMemberResponse {
    items: string[];
}

export class ProjectMemberCreate extends BaseRequest {
    projectName: string;
    userName: string;
    role: string;
}
